src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadataWorker.cs

```diff
@@ -342,6 +342,11 @@ private static ExtractMetadataInputModel ValidateInput(ExtractMetadataInputModel
                     .Where(s => s is PortableExecutableReference)
                     .Select(s => ((PortableExecutableReference)s).FilePath));
                 FillProjectDependencyGraph(projectCache, projectDependencyGraph, project);
+                // duplicate project references will fail Project.GetCompilationAsync
+                if (project.ProjectReferences.GroupBy(r => r.ProjectId).Any(g => g.Count() > 1))
+                {
+                    projectCache[path] = project.WithProjectReferences(project.ProjectReferences.Distinct());
```

Comment:
is it true that the equality of ProjectReferences.ProjectId means the equality of ProjectReference?